internal/fs/inode/file.go

```diff
@@ -545,10 +547,12 @@ func (f *FileInode) Sync(ctx context.Context) (err error) {
 
 	// Special case: a precondition error means we were clobbered, which we treat
 	// as being unlinked. There's no reason to return an error in that case.
-	var preconditionErr *gcs.PreconditionError
-	if errors.As(err, &preconditionErr) {
-		err = nil
-		return
+	var gErr *googleapi.Error
+	if errors.As(err, &gErr) {
```

Comment:
does it work for jacobsa flow as well?

Reply:
In jacobsa flow it is not throwing 412 error

Reply:
fixed